### ISSUE TITLE
CA1507 suppressor (MAS0004) uses `return` instead of `continue`, so it stops at the first unrelated diagnostic

### ISSUE THREAD

**Issue by meziantou** [meziantou/Meziantou.Analyzer#1337]
`ReportSuppressions` iterates `context.ReportedDiagnostics` but uses `return` — not `continue` — for the three "not applicable" cases:

https://github.com/meziantou/Meziantou.Analyzer/blob/main/src/Meziantou.Analyzer/Suppressors/CA1507SerializationPropertyNameSuppressor.cs#L18-L38

`context.ReportedDiagnostics` contains **every** CA1507 in the compilation, and the vast majority are not inside an attribute — so `node.FirstAncestorOrSelf<AttributeSyntax>()` is null and the method exits before ever reaching the `[JsonProperty]` one.

### Repro

```csharp
internal class Test
{
    public void Foo(string name) => throw new System.ArgumentException("dummy", "name"); // CA1507, not in an attribute

    [Newtonsoft.Json.JsonProperty("Bar")]
    public int Bar { get; set; } // the CA1507 MAS0004 is supposed to suppress
}
```

If the first diagnostic is processed first, the second is never reached and CA1507 is reported.

### Why it matters

MAS0004 is effectively non-functional in any real project — every project has at least one non-attribute CA1507 — and whether it works at all depends on the ordering of `ReportedDiagnostics`, which is not guaranteed. The three existing tests each contain exactly one CA1507, so the bug is invisible to the suite.

### Suggested fix

Replace the three `return` statements with `continue`. While there, hoist `GetBestTypeByMetadataName("Newtonsoft.Json.JsonPropertyAttribute")` out of the loop and bail early when it is null.

A test with two CA1507 diagnostics in one file, the attribute one second, would lock this in.

---
<sub>Found during a whole-project review. Reproduced against `Meziantou.Analyzer.Test.roslyn5.9.csproj` at commit 22602bed.</sub>
